src/main/java/org/folio/service/dataimport/handlers/CreateOrderEventHandler.java

```diff
@@ -0,0 +1,325 @@
+import static org.folio.DataImportEventTypes.DI_ORDER_CREATED;
+import static org.folio.orders.utils.HelperUtils.ORDER_CONFIG_MODULE_NAME;
+import static org.folio.orders.utils.HelperUtils.PO_LINES_LIMIT_PROPERTY;
+import static org.folio.rest.jaxrs.model.ProfileSnapshotWrapper.ContentType.ACTION_PROFILE;
+
+@Component
+public class CreateOrderEventHandler implements EventHandler {
+
+  private static final Logger LOGGER = LogManager.getLogger();
+  private static final String PAYLOAD_HAS_NO_DATA_MSG =
+    "Failed to handle event payload, cause event payload context does not contain MARC_BIBLIOGRAPHIC data";
+
+  public static final String PERMISSIONS_KEY = "OKAPI_PERMISSIONS";
+  public static final String USER_ID_KEY = "USER_ID";
+  public static final String OKAPI_PERMISSIONS_HEADER = "X-Okapi-Permissions";
+  static final String POL_LIMIT_RULE_NAME = "overridePoLinesLimit";
+  private static final String ORDER_FIELD = "po";
+  private static final String PO_LINES_FIELD = "poLine";
+  private static final String MAPPING_RESULT_FIELD = "order";
+  private static final String INSTANCE_ID_FIELD = "id";
+  private static final String ORDER_STATUS_FIELD = "workflowStatus";
+  private static final String POL_ACTIVATION_DUE_FIELD = "activationDue";
+  private static final String POL_ERESOURCE_FIELD = "eresource";
+  private static final String POL_PHYSICAL_FIELD = "physical";
+  private static final String POL_CREATE_INVENTORY_FIELD = "createInventory";
+  private static final String ORDER_LINES_KEY = "ORDER_LINES";
+  private static final String RECORD_ID_HEADER = "recordId";
+  private static final String JOB_PROFILE_SNAPSHOT_ID_KEY = "JOB_PROFILE_SNAPSHOT_ID";
+  private static final String ID_UNIQUENESS_ERROR_MSG = "duplicate key value violates unique constraint";
+  private static final String PROFILE_SNAPSHOT_NOT_FOUND_MSG = "JobProfileSnapshot was not found by profileSnapshotId '%s'";
+
+  private final PurchaseOrderHelper purchaseOrderHelper;
+  private final PurchaseOrderLineHelper poLineHelper;
+  private final ConfigurationEntriesService configurationEntriesService;
+  private final IdStorageService idStorageService;
+  private final JobProfileSnapshotCache jobProfileSnapshotCache;
+
+  @Autowired
+  public CreateOrderEventHandler(PurchaseOrderHelper purchaseOrderHelper, PurchaseOrderLineHelper poLineHelper,
+                                 ConfigurationEntriesService configurationEntriesService, IdStorageService idStorageService,
+                                 JobProfileSnapshotCache jobProfileSnapshotCache) {
+    this.purchaseOrderHelper = purchaseOrderHelper;
+    this.poLineHelper = poLineHelper;
+    this.configurationEntriesService = configurationEntriesService;
+    this.idStorageService = idStorageService;
+    this.jobProfileSnapshotCache = jobProfileSnapshotCache;
+  }
+
+  @Override
+  public CompletableFuture<DataImportEventPayload> handle(DataImportEventPayload dataImportEventPayload) {
```

Comment:
We should not use java.util.CompletableFuture, please use Verx Future

Reply:
I see now, it is interface from data-import-processing-core, and this library uses CompletableFuture

Reply:
yes, it is required by the interface contract, but all async results of the handler and di-processing-core execution are handled by vertx thread